service/controller/chart/v1/resource/tillermigration/resource.go

```diff
@@ -0,0 +1,137 @@
+package tillermigration
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/giantswarm/apiextensions/pkg/clientset/versioned"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+const (
+	Name = "tillermigrationv1"
+)
+
+// Config represents the configuration used to create a new tiller resource.
+type Config struct {
+	// Dependencies.
+	G8sClient versioned.Interface
+	K8sClient kubernetes.Interface
+	Logger    micrologger.Logger
+}
+
+// Resource implements the tillermigration resource.
```

Comment:
I find these kind of comments utterly useless. It does not explain anything. We should try to write more comments explaining why a certain piece of software implements something, and not solely what it implements.

Reply:
Since this is the migration resource would be gone at some points, I will delete comments to not take too much time on it.